aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -172,6 +172,22 @@ pub struct AptosVM {
 
 impl AptosVM {
     pub fn new(resolver: &impl AptosMoveResolver) -> Self {
+        Self::new_impl(resolver, resolver.is_delayed_field_optimization_capable())
+    }
+
+    pub(crate) fn new_with_delayed_fields(resolver: &impl AptosMoveResolver) -> Self {
+        Self::new_impl(resolver, true)
```

Comment:
Should we assert `resolver.is_delayed_field_optimization_capable())`

Reply:
Use a single constructor instead